### PR TITLE
task_DEV-3

### PR DESCRIPTION
**Pull request by Zmitsier-Sakhonchyk**

### CODE REVIEW

File: task_DEV-3.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
+﻿using System;
+
+namespace FibonacciNumbers
+{
+    class Program
+    {
+        static void Main()
+        {
+            Console.WriteLine("Enter a number:");
+            string numberString = Console.ReadLine();
```
Comment (IgorKheidorov) [inline task_DEV-3.cs]: Use const

File: task_DEV-3.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                Environment.Exit(0);
+            }
+            if (number < 0)
+            {
+                Console.WriteLine("\nInput number is negative.");
+                Console.ReadKey();
+                Environment.Exit(0);
+            }
+
+            int fibonacciN1 = 0;
```
Comment (IgorKheidorov) [inline task_DEV-3.cs]: use const

File: task_DEV-3.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            try
+            {
+                number = Int32.Parse(numberString);
+            }
+            catch (FormatException e)
+            {
+                Console.Write("\nFormatException: ");
+                Console.WriteLine(e.Message);
+                Console.ReadKey();
+                Environment.Exit(0);
```
Comment (IgorKheidorov) [inline task_DEV-3.cs]: do not use system exit, ask for re-input value

File: task_DEV-3.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            }
+            catch (OverflowException e)
+            {
+                Console.Write("\nOverflowException: ");
+                Console.WriteLine("Number is too large.");
+                Console.ReadKey();
+                Environment.Exit(0);
```
Comment (IgorKheidorov) [inline task_DEV-3.cs]: do not use system exit, ask for re-input value

File: task_DEV-3.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
+﻿using System;
```
Comment (IgorKheidorov) [inline task_DEV-3.cs]: Add all files necessary for compilation (cproj, app.config, etc)

File: task_DEV-3.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            do
+            {
+                Console.WriteLine("Enter a number:");
+                string numberString = Console.ReadLine();                
+                try
+                {
+                    number = Int32.Parse(numberString);
+                    repeat = false;
+                }
+                catch (FormatException e)
```
Comment (IgorKheidorov) [inline task_DEV-3.cs]: You have similar processing for all exceptions, try to combine it not to have similar code fragments. Do not repeat youself

File: task_DEV-3.cs [hunk 7]
```diff
@@ -0,0 +1,63 @@
+﻿using System;
```
Comment (IgorKheidorov) [inline task_DEV-3.cs]: Are you sure all necessary files are added? Make temp clone from git and try to build project.

**Review by IgorKheidorov — COMMENTED** [review]
Remove all binary files from pull request, leave only necessary ones, then I'll review other issues

File: task_DEV-3/task_DEV-3.sln [hunk 8]
```diff
@@ -0,0 +1,22 @@
+﻿
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3.sln]: One solution for all projects, not create solution for separate task

File: task_DEV-3/task_DEV-3/task_DEV-3.cs [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+            int fibonacciN1 = 0;
+            int fibonacciN2 = 1;
+            int fibonacciN;
+            do
+            {
+                fibonacciN = fibonacciN1 + fibonacciN2;
+                fibonacciN1 = fibonacciN2;
+                fibonacciN2 = fibonacciN;
+            } while (fibonacciN < number);
+            if ((number == fibonacciN) || (number == 0))
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/task_DEV-3.cs]: try to simplify

Comment (IgorKheidorov) [reply task_DEV-3/task_DEV-3/task_DEV-3.cs]: Good decision simply return (number == fibonacciN) || (number == 0)

File: task_DEV-3/task_DEV-3/task_DEV-3.cs [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+    }
+    
+    class Program
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/task_DEV-3.cs]: Bad class name, try to rename like FibonacciNumberSearchTool

File: task_DEV-3/task_DEV-3/task_DEV-3.cs (hunk 9 above)
Comment (Zmitsier-Sakhonchyk) [reply task_DEV-3/task_DEV-3/task_DEV-3.cs]: I don't understand.

File: task_DEV-3/task_DEV-3.sln (hunk 8 above)
Comment (Zmitsier-Sakhonchyk) [reply task_DEV-3/task_DEV-3.sln]: Should I delete this one?

File: task_DEV-3/task_DEV-3/task_DEV-3.cs [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+        public static bool ShowMessage (string exceptionType, string message)
+        {
+            Console.Write(exceptionType);
+            Console.WriteLine(message);
+            Console.WriteLine("\nPlease, repeat number entry.\n");
+            return true;
+        }
+    }
+    
+    public static class FibonacciNumbers
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/task_DEV-3.cs]: Why static? bad idea

File: task_DEV-3/task_DEV-3/task_DEV-3.cs [hunk 12]
```diff
@@ -0,0 +1,73 @@
+﻿using System;
+
+namespace FibonacciNumbers
+{
+    public static class ExceptionMessage
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/task_DEV-3.cs]: Static is a bad idea

**Review by IgorKheidorov — APPROVED** [review]
Totally exceptable

**Pull request merged**